edkrepo/common/common_repo_functions.py

```diff
@@ -908,17 +908,19 @@ def _list_patchset_ancestors(manifest, patchset):
 
 def _checkout_parent(patchset, name, repo, parent_patchsets):
     if parent_patchsets:
-        base_patchset_parent_sha = parent_patchsets[-1].parent_sha
-        if base_patchset_parent_sha in repo.tags:
-            # tag names are prioritized over branch names
-            repo.git.checkout("refs/tags/{}".format(base_patchset_parent_sha), b=name)
-        else:
-            repo.git.checkout(base_patchset_parent_sha, b=name)
-    elif patchset.parent_sha in repo.tags:
+        base_parent_sha = parent_patchsets[-1].parent_sha
+        if (base_parent_sha in repo.heads and

```

Comment:
This check can still be bypassed if the manifest file PatchSet uses a remote branch name as a parentSha.

Edk2 PatchSet example:  parentSha="origin/UDK2017" will still complete successfully.

Is that acceptable?

Reply:
We should probably incorporate a check for this somewhere else if it hasn't already been added. I'd need to look through the existing code. Seems like a business logic rule that should maybe be placed earlier in the entire dynamic branch creation process. I think for now it might be okay to leave it as is, and we may be able to address it in a future issue.

Reply:
@kevinsun49 please file an issue to track this